src/python/pants/engine/internals/parametrize.py

```diff
@@ -166,42 +166,54 @@ def expand(
         except Exception as e:
             raise Exception(f"Failed to parametrize `{address}`:\n  {e}") from e
 
+        parameters = address.parameters
+        non_parametrized = tuple(
+            (field_name, field_value)
+            for field_name, field_value in fields.items()
+            if not isinstance(field_value, Parametrize)
+        )
+        non_parametrized_keys = {k for k, _ in non_parametrized}
```

Comment:
we could do the union here, so we can check it later to only conditionally create a new address when there are no parameterizations ;)
```suggestion
        unparametrize_keys = {k for k, _ in non_parametrized} & parameters.keys()
```